File: week_10/c_sharp_course/week_1/day_5/solid/code/single_responsibility_principle/after/Athlete.cs
public class Athlete {
  private AthleteConsoleLog log;
  private int totalDistance;

  public Athlete()
  {
    this.log = new AthleteConsoleLog();
  }

  public void Run(int distance)
  {
    totalDistance += distance;
    this.log.Distance(distance, totalDistance);
  }
}

File: week_11/day2/bottle_end_code/athlete.js
var Athlete = function () {
  this.hydration = 100;
  this.distanceCovered = 0;
}


Athlete.prototype.run = function(distance) {
  if (this.isDehydrated()) return;
  this.dehydrate();
  this.distanceCovered += distance;
}

Athlete.prototype.isDehydrated = function() {
  return this.hydration === 0;
}

Athlete.prototype.dehydrate = function() {
  if (this.hydration <= 10) {
    this.hydration = 0;
  }
  else {
    this.hydration -= 10;
  }
}

Athlete.prototype.drink = function(bottle) {
  this.hydration += bottle.drink();
}


module.exports = Athlete;
